File: 07-Encordados-JPA-Hibernate-Java/src/main/java/controllers/MutuaController.java
package controllers;

import models.Raqueta;
import models.Tenista;
import repositories.raquetas.RaquetasRepository;
import repositories.tenistas.TenistasRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import java.util.logging.Logger;

public class MutuaController {
    private final Logger logger = Logger.getLogger(MutuaController.class.getName());

    // Mis dependencias
    private final RaquetasRepository raquetasRepository;
    private final TenistasRepository tenistasRepository;

    public MutuaController(RaquetasRepository raquetasRepository, TenistasRepository tenistasRepository) {
        this.raquetasRepository = raquetasRepository;
        this.tenistasRepository = tenistasRepository;
    }

    // Tenistas
    public List<Tenista> getTenistas() {
        logger.info("Obteniendo Tenistas");
        return tenistasRepository.findAll();
    }

    public Tenista createTenista(Tenista tenista) {
        logger.info("Creando Tenista");
        return tenistasRepository.save(tenista);
    }

    public Optional<Tenista> getTenistaById(UUID uuid) {
        logger.info("Obteniendo Tenista con uuid: " + uuid);
        return tenistasRepository.findById(uuid);
    }

    public Tenista updateTenista(Tenista tenista) {
        logger.info("Actualizando Tenista con uuid: " + tenista.getUuid());
        return tenistasRepository.save(tenista);
    }

    public Boolean deleteTenista(Tenista tenista) {
        logger.info("Eliminando Tenista con uuid: " + tenista.getUuid());
        return tenistasRepository.delete(tenista);
    }

    // Raquetas
    public List<Raqueta> getRaquetas() {
        logger.info("Obteniendo Raquetas");
        return raquetasRepository.findAll();
    }

    public Raqueta createRaqueta(Raqueta raqueta) {
        logger.info("Creando Raqueta");
        return raquetasRepository.save(raqueta);
    }

    public Optional<Raqueta> getRaquetaById(UUID uuid) {
        logger.info("Obteniendo Raqueta con uuid: " + uuid);
        return raquetasRepository.findById(uuid);
    }

    public Raqueta updateRaqueta(Raqueta raqueta) {
        logger.info("Actualizando Raqueta con uuid: " + raqueta.getUuid());
        return raquetasRepository.save(raqueta);
    }

    public Boolean deleteRaqueta(Raqueta raqueta) {
        logger.info("Eliminando Raqueta con uuid: " + raqueta.getUuid());
        return raquetasRepository.delete(raqueta);
    }
}

File: 06-Encordados-Exposed-Kotlin/src/main/kotlin/controllers/MutuaController.kt
package controllers

import models.Raqueta
import models.Tenista
import mu.KotlinLogging
import repositories.raqueta.RaquetasRepository
import repositories.tenista.TenistasRepository
import java.util.*

private val logger = KotlinLogging.logger {}

class MutuaController(
    private val raquetasRepository: RaquetasRepository,
    private val tenistasRepository: TenistasRepository
) {

    // TENISTAS

    fun getTenistas(): List<Tenista> {
        logger.info("Obteniendo Tenistas")
        return tenistasRepository.findAll()
    }

    fun createTenista(tenista: Tenista): Tenista {
        logger.debug { "Creando tenista $tenista" }
        tenistasRepository.save(tenista)
        return tenista
    }

    fun getTenistaById(uuid: UUID): Tenista? {
        logger.debug { "Obteniendo tenista con uuid $uuid" }
        return tenistasRepository.findById(uuid)
    }

    fun updateTenista(tenista: Tenista) {
        logger.debug { "Updating tenista $tenista" }
        tenistasRepository.save(tenista)
    }

    fun deleteTenista(it: Tenista): Boolean {
        logger.debug { "Borrando tenista $it" }
        return tenistasRepository.delete(it)
    }

    // RAQUETAS
    fun getRaquetas(): List<Raqueta> {
        logger.info("Obteniendo Raquetas")
        return raquetasRepository.findAll()
    }

    fun createRaqueta(raqueta: Raqueta): Raqueta {
        logger.debug { "Creando raqueta $raqueta" }
        raquetasRepository.save(raqueta)
        return raqueta
    }

    fun getRaquetaById(uuid: UUID): Raqueta? {
        logger.debug { "Obteniendo raqueta con uuid $uuid" }
        return raquetasRepository.findById(uuid)
    }

    fun updateRaqueta(raqueta: Raqueta): Raqueta {
        logger.debug { "Actualizando $raqueta" }
        return raquetasRepository.save(raqueta)
    }

    fun deleteRaqueta(it: Raqueta): Boolean {
        logger.debug { "Borrando raqueta $it" }
        return raquetasRepository.delete(it)
    }
}